Ginger/GingerCore/ALM/QtestCore.cs

```diff
@@ -0,0 +1,657 @@
+                connObj.Configuration.MyAPIConfig = new QTestApiClient.QTestClientConfig();
+                connObj.Configuration.AccessToken = response.AccessToken;
+                connObj.Configuration.ApiKey.Add("Authorization", response.AccessToken);
+                connObj.Configuration.ApiKeyPrefix.Add("Authorization", response.TokenType);
+                System.Diagnostics.Trace.WriteLine("Authentication Successful");
+                return true;
+            }           
+            catch (Exception ex)
+            {
+                System.Diagnostics.Trace.WriteLine("Exception in AuthenticateUser(): Authentication Failed: " + ex.Message);
+
+                connObj = null;
+                return false;
+            }
+        }
+
+        public override bool ConnectALMProject()
+        {
+            ALMCore.DefaultAlmConfig.ALMProjectName = ALMCore.DefaultAlmConfig.ALMProjectKey;
+            return true;
+        }
+
+        public override Boolean IsServerConnected()
+        {
+            // return QCConnect.IsServerConnected;
+            return true;
+        }
+
+        public override void DisconnectALMServer()
+        {
+            // QCConnect.DisconnectQCServer();
+        }
+
+        public override List<string> GetALMDomains()
+        {
+            // return QCConnect.GetQCDomains();
+            return new List<string>();
+        }
+
+        public override Dictionary<string,string> GetALMDomainProjects(string ALMDomain)
+        {
+            projectsApi = new QTestApi.ProjectApi(connObj.Configuration);
+            List<QTestApiModel.ProjectResource> projectList =  projectsApi.GetProjects("descendents", true);
+            return projectList.ToDictionary(f => f.Id.ToString(), f => f.Name);
+        }
+
+        public override bool DisconnectALMProjectStayLoggedIn()
+        {
+            // return QCConnect.DisconnectQCProjectStayLoggedIn();
+            return true;
+        }
+
+        public override ObservableList<ActivitiesGroup> GingerActivitiesGroupsRepo
+        {
+            get { return ImportFromQtest.GingerActivitiesGroupsRepo; }
+            set { ImportFromQtest.GingerActivitiesGroupsRepo = value; }
+        }
+
+        public override ObservableList<Activity> GingerActivitiesRepo
+        {
+            get { return ImportFromQtest.GingerActivitiesRepo; }
+            set { ImportFromQtest.GingerActivitiesRepo = value; }
+        }
+       
+        public override ObservableList<ExternalItemFieldBase> GetALMItemFields(BackgroundWorker bw, bool online, ALM_Common.DataContracts.ResourceType resourceType)
+        {
+            ConnectALMServer();
+            fieldApi = new QTestApi.FieldApi(connObj.Configuration);
+            ObservableList<ExternalItemFieldBase> fields = new ObservableList<ExternalItemFieldBase>();
+
+            if (resourceType == ALM_Common.DataContracts.ResourceType.ALL)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)